azure-templates/stages.yml

```diff
@@ -48,6 +52,53 @@ parameters:
 
 # Note: If custom build steps are needed for a particular custom device, copy this entire template into the azure-pipeline and add steps before/after as needed
 stages:
+  - stage: DiffVIs${{ parameters.stageName }}
+    dependsOn: []
+    condition: and(eq(variables['Build.Reason'], 'PullRequest'), eq('${{ parameters.disableDiff }}', false))
+    pool:
+      name: DevCentral-VeriStand-CustomDevices
+      demands:
+        - agent.os -equals Windows_NT
+    jobs:
+      - job: Find64bitVersion
+        steps:
+          - powershell: |
+              $lvVersionsToBuild = '${{ convertToJson(parameters.lvVersionsToBuild) }}' | ConvertFrom-Json
+              ForEach ($lvVersionToBuild in $lvVersionsToBuild)
+              {
+                if ($lvVersionToBuild.bitness -eq "64bit")
+                {
+                  $bitness = $lvVersionToBuild.bitness
+                  $version = $lvVersionToBuild.version
+                  Write-Host "##vso[task.setvariable variable=bitness;isOutput=true;]$bitness"
+                  Write-Host "##vso[task.setvariable variable=version;isOutput=true;]$version"
+                }
+              }
+            name: FindLast64bitVersion
+      - job: DiffVIsforPR
+        displayName: Diff VIs for PR
+        dependsOn: ['Find64bitVersion']
+        variables:
+          - group: niveristand-custom-devices
+          - name: diffBitness
+            value: $[ dependencies.Find64bitVersion.outputs['FindLast64bitVersion.bitness'] ]
+          - name: diffVersion
+            value: $[ dependencies.Find64bitVersion.outputs['FindLast64bitVersion.version'] ]
+        steps:
+          - template: steps-pre-build.yml # Configure variables, check out repos, Clear cache
```

Comment:
option:  Could add a conditional similar to "disableDiff" that disables the cache clearing, since that is the most time consuming item here, and skipping that step might be desired in other places as well

Reply:
After talking with @papowerNI decided to defer large-scale rework of the stages and template files for later tech debt cleanup.